website/static/js/profile.js

```diff
@@ -468,7 +468,44 @@ var NameViewModel = function(urls, modes, preventUnsaved, fetchCallback) {
         return suffix;
     };
 
+    self.fullNameParsed = function() {
```

Comment:
Here's my recommendation to simplify: 
1. Check the last part to see if it is a suffix (new function). If it is, pop it off and assign to suffix
2. Pop off the first piece and assign to first name (if len === 0 return nameobj)
3. Pop off last piece and assign to last name (if len === 0 return nameobj)
4. Assign remaining pieces (however many there are) to middle name, space separated